code/modules/mob/living/silicon/robot/robot_model.dm

```diff
@@ -158,6 +158,9 @@
 			var/obj/item/gun/energy/gun = module
 			if(!gun.chambered)
 				gun.recharge_newshot() //try to reload a new shot.
+		else if(istype(module, /obj/item/soap))
```

Comment:
This should be done in `/obj/item/robot_model/janitor/respawn_consumable`, taking the coeff into account. That way the soap slowly regens